### PR TITLE
[PT-4990] Add CI

### PR DESCRIPTION
**Pull request by muripic**
## Context and changes made

This PR adds a CI pipeline to fake-kafka that runs rubocop and rspec.

The goal is to improve maintainability of this gem to ensure compatibility with Ruby 3 and also be in a better position to accept external contributions, since it's an open source library.

Since this is an open source gem and external users don’t have access to CircleCI, we decided to implement the pipeline using GitHub Actions.

Link to task: https://catawiki.atlassian.net/browse/PT-4990

If approved, I plan to add branch protection settings that do not allow a PR to be merged if these checks do not pass.

~~I added the tag `version:minor` because we are adding compatibility with Ruby 3.0 and will ensure it from now on, but we can follow an alternative approach 😄~~ Just noticed that the previous release 0.0.3 was not published either, so I'm changing the tag to `version:no release`

### CODE REVIEW

**Comment by dhruvCW** [comment]
>I added the tag version:minor because we are adding compatibility with Ruby 3.0 and will ensure it from now on, but we can follow an alternative approach 😄

didn't @sarcilav mention that we can only do patch updates, otherwise it might conflict with the original library ?

File: .rubocop_todo.yml
Comment (dhruvCW) [inline .rubocop_todo.yml:1]: are we planning to fix rubocop in a future PR ? also are we not pinning the ruby version for rubocop ?

File: .github/workflows/run-tests-and-checks.yml
Comment (dhruvCW) [inline .github/workflows/run-tests-and-checks.yml:19]: shouldn't we do `bundle install` ?

File: .github/workflows/run-tests-and-checks.yml
Comment (dhruvCW) [inline .github/workflows/run-tests-and-checks.yml:36]: why is this required ?

**Review by dhruvparanjape — COMMENTED** [review]
couple of comments, but looks good overall 🍻 great effort :)

**Comment by muripic** [comment]
> > I added the tag version:minor because we are adding compatibility with Ruby 3.0 and will ensure it from now on, but we can follow an alternative approach 😄
> 
> didn't @sarcilav mention that we can only do patch updates, otherwise it might conflict with the original library ?

That applies to our backport of ruby-kafka, this one is a different gem created at Catawiki 😄

**Comment by dhruvCW** [comment]
>That applies to our backport of ruby-kafka, this one is a different gem created at Catawiki 😄

I misread the title 🤦 , my bad.

File: .rubocop_todo.yml
Comment (muripic) [reply .rubocop_todo.yml:1]: I decided not to pin the Ruby version for Rubocop since we're supporting 3 different versions, but I'm open to suggestions 😄

Comment (dhruvCW) [reply .rubocop_todo.yml:1]: 3 different rubocop files (all inheriting from a common base) ?

File: Gemfile.lock
Comment (muripic) [reply Gemfile.lock:4]: That's an option, bundler also recommends to leave it in some cases as explained [here](https://bundler.io/guides/faq.html#using-gemfiles-inside-gems). Even though it's true that at Catawiki we usually remove it, I decided to keep it here since it was originally included.

I'm not against going for it 😄  But I think it'd be better if it's done in a separate PR.

File: fake-kafka.gemspec
Comment (muripic) [reply fake-kafka.gemspec:27]: In this case I just followed the original approach because I wanted to limit my changes as much as possible to adding CI with the basic checks, but please feel free to tackle those issues in a separate PR 😄

File: .rubocop_todo.yml
Comment (muripic) [reply .rubocop_todo.yml:1]: Also decided not to go for the rubocop fixes here since IMO they are out of scope for this PR, and they would pollute these changes and make it very hard to review.
IMO it should definitely be done in future PRs, and it shouldn't be too much work since there are many autocorrectable fixes and it's a small gem 😄 But I'll leave that up to the new maintainer team 😄

File: .github/workflows/run-tests-and-checks.yml
Comment (muripic) [reply .github/workflows/run-tests-and-checks.yml:19]: I originally wanted to avoid running another `bundle install` but then discovered that we can cache it and it doesn't take too long anyway. And also it will help us keep consistency with versions, so I'll make the change 😄

File: .rubocop_todo.yml
Comment (muripic) [reply .rubocop_todo.yml:1]: > 3 different rubocop files (all inheriting from a common base) ?

Could be 🤔
I've just checked the[ rubocop docs on the target Ruby version](https://docs.rubocop.org/rubocop/configuration.html#setting-the-target-ruby-version) to be sure and if I understood correctly, we're supposed to specify the oldest version supported by the gem. It should also be inferred from the gemspec, but I can add it to the rubocop config to make it more explicit 😄

Comment (muripic) [reply .rubocop_todo.yml:1]: done

File: Gemfile.lock
Comment (tdyas) [reply Gemfile.lock:4]: Maybe remove it then in a pre-work PR (which we can approve quickly)? That would simplify this PR's diff, right?

File: .github/workflows/run-tests-and-checks.yml
Comment (muripic) [reply .github/workflows/run-tests-and-checks.yml]: done

File: .github/workflows/run-tests-and-checks.yml
Comment (muripic) [reply .github/workflows/run-tests-and-checks.yml]: done

File: Gemfile.lock
Comment (muripic) [reply Gemfile.lock:4]: Sure, if there is already a general agreement on removing the `Gemfile.lock` among the new maintainers, I'll be happy to do it 😄

File: .github/workflows/run-tests-and-checks.yml
Comment (dhruvCW) [inline .github/workflows/run-tests-and-checks.yml:18]: I believe this flag runs `bundle install` automatically according to this statement `bundler-cache: true # runs 'bundle install' and caches installed gems automatically` from https://github.com/ruby/setup-ruby#matrix-of-ruby-versions

so we can exclude teh install deps step.

File: Gemfile.lock
Comment (muripic) [reply Gemfile.lock:4]: Done: https://github.com/catawiki/fake-kafka/pull/22

File: .github/workflows/run-tests-and-checks.yml
Comment (muripic) [reply .github/workflows/run-tests-and-checks.yml:18]: Good point, I hadn't seen that 💯 Now when I rebase this PR on top of the other branch I'll make that change too 🙇‍♀️

Comment (muripic) [reply .github/workflows/run-tests-and-checks.yml]: done

**Review by dhruvparanjape — DISMISSED** [review]
🚀  LGTM

**Comment by muripic** [comment]
@dhruvCW @tdyas just FYI I switched back the label to `version: no release` since I've just noticed that there are some issues with the publishing of the gem due to access to the RubyGems repository. There are some more details here:  https://github.com/catawiki/fake-kafka/pull/14
From Platform team we do not have access to this and ATM we do not have the bandwidth to follow up on that 😞
Is it ok if I merge it like this and you follow up on this with #support-opensource as the new maintainers? 🙏

**Comment by dhruvCW** [comment]
>you follow up on this with #support-opensource as the new maintainers? 🙏

we are the new maintainers ?

>Is it ok if I merge it like this 

the only "gem" change is the ruby version restriction, so if we remove that with a TODO to follow up I am okay with merging it as `no_release`

**Comment by muripic** [comment]
> > Is it ok if I merge it like this
> 
> the only "gem" change is the ruby version restriction, so if we remove that with a TODO to follow up I am okay with merging it as `no_release`

👍 I'll do that then so we can merge it

**Comment by muripic** [comment]
> > you follow up on this with #support-opensource as the new maintainers? 🙏
> 
> we are the new maintainers ?
> 
> > Is it ok if I merge it like this
> 
> the only "gem" change is the ruby version restriction, so if we remove that with a TODO to follow up I am okay with merging it as `no_release`

Done and readded the corresponding rubocop TODO.

**Review by dhruvparanjape — APPROVED** [review]

**Pull request merged**
